system-test/spanner.js

```diff
@@ -2198,6 +2201,305 @@ describe('Spanner', function() {
             });
           });
         });
+
+        describe.only('structs', function() {
+          it('should bind a simple struct', function(done) {
+            var query = {
+              sql: 'SELECT @structParam.userf, @p4',
+              params: {
+                structParam: Spanner.struct({
+                  userf: 'bob',
+                  threadf: Spanner.int(1),
+                }),
+                p4: Spanner.int(10),
+              },
+            };
+
+            database.run(query, function(err, rows) {
+              assert.ifError(err);
+
+              var row = rows[0].toJSON();
+              assert.strictEqual(row.userf, 'bob');
+
+              done();
+            });
+          });
+
+          it('should bind null structs', function(done) {
+            var query = {
+              sql: 'SELECT @structParam.userf',
+              params: {
+                structParam: null,
+              },
+              types: {
+                structParam: {
+                  type: 'struct',
+                  fields: [
+                    {
+                      name: 'userf',
+                      type: 'string',
+                    },
+                    {
+                      name: 'threadf',
+                      type: 'int64',
+                    },
+                  ],
+                },
+              },
+            };
+
+            database.run(query, function(err, rows) {
+              assert.ifError(err);
+
+              var row = rows[0].toJSON();
+              assert.strictEqual(row.userf, null);
+
+              done();
+            });
+          });
+
+          it('should bind nested structs', function(done) {
+            var query = {
+              sql: 'SELECT @structParam.structf.nestedf',
+              params: {
+                structParam: Spanner.struct({
+                  structf: Spanner.struct({
+                    nestedf: 'bob',
+                  }),
+                }),
+              },
+            };
+
+            database.run(query, function(err, rows) {
+              assert.ifError(err);
+
+              var row = rows[0].toJSON();
+              assert.strictEqual(row.nestedf, 'bob');
+
+              done();
+            });
+          });
+
+          it('should bind null nested structs', function(done) {
+            var query = {
+              sql: 'SELECT @structParam.structf.nestedf',
+              params: {
+                structParam: null,
+              },
+              types: {
```

Comment:
what happens if types is not specified and the value for structParams is null?

Reply:
An exception would be thrown.